stellargraph/layer/gcn.py

```diff
@@ -0,0 +1,258 @@
+        activations,
+        generator,
+        bias=True,
+        dropout=0.0,
+        normalize=regularizers.l2(5e-4),
+    ):
+
+        """
+
+        Args:
+            layer_sizes: list of output sizes of GCN layers in the stack
+            activations: list of activations applied to each layer's output
+            generator: an instance of FullBatchNodeGenerator class constructed on the graph of interest
+            bias: toggles an optional bias in GCN layers
+            dropout: dropout rate applied to input features of each GCN layer
+            normalize: normalization applied to the final output features of the GCN layers stack
+        """
+
+        if not isinstance(generator, FullBatchNodeGenerator):
+            raise TypeError("Generator should be a instance of FullBatchNodeGenerator")
+
+        assert len(layer_sizes) == len(activations)
+
+        self.layer_sizes = layer_sizes
+        self.activations = activations
+        self.bias = bias
+        self.dropout = dropout
+        self.normalize = normalize
+        self.generator = generator
+        self.support = 1
+
+        # Initialize a stack of GCN layers
+        self._layers = []
+        for l, a in zip(self.layer_sizes, self.activations):
+            self._layers.append(Dropout(self.dropout))
+            self._layers.append(
+                GraphConvolution(
+                    l,
+                    self.support,
+                    activation=a,
+                    use_bias=self.bias,
+                    kernel_regularizer=self.normalize,
+                )
+            )
+
+    def __call__(self, x: List):
+        """
+        Apply a stack of GCN layers
+
+        Args:
+            x (list of Tensor): input features
+
+        Returns:
+            Output tensor
+        """
+
+        H = x[0]
+        suppG = x[1:]
+
+        for layer in self._layers:
+            if isinstance(layer, GraphConvolution):
+                # It is a GCN layer
+                H = layer([H] + suppG)
+            else:
+                # layer is a Dropout layer
+                H = layer(H)
+
+        return H
+
+    def node_model(self):
+        """
+        Builds a GCN model for node prediction
+        Returns:
+            tuple: (x_inp, x_out) where ``x_inp`` is a Keras input tensor
+            for the specified GCN model and ``x_out`` is a Keras tensor for the GCN model output.
+        """
+
+        x_in = Input(shape=(self.generator.features.shape[1],))
+
+        filter = self.generator.kwargs.get("filter", "localpool")
+        if filter == "chebyshev":
+            self.support = self.generator.kwargs.get("max_degree", 2)
+            suppG = [
+                Input(shape=(None, None), batch_shape=(None, None), sparse=True)
+                for _ in range(self.support)
+            ]
+        else:
+            suppG = [Input(shape=(None, None), batch_shape=(None, None), sparse=True)]
+
+        x_out = self([x_in] + suppG)
```

Comment:
it might be better, in terms of readability and efficiency, to modify lines 248-249 with
```
x_inp = [x_in] + suppG
x_out = self(x_inp)
return x_inp, x_out
```
This way the list creation `[x_in} + suppG` is done only once, instead of twice.

Reply:
https://github.com/stellargraph/stellargraph/blob/feature/gcn/stellargraph/layer/gcn.py#L252